src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -2700,6 +2706,40 @@ var projectUniqueNamesForBuildIntToUpdate
             await OpenReadmeFile(buildIntegratedProject, nuGetProjectContext, token);
         }
 
+        /// <summary>
+        /// Function used to display errors and warnings at the end of Install operation.
+        /// The errors and warnings are read from the assets file based on preview restore.
+        /// </summary>
+        /// <param name="lockFile">LockFile generated by preview restore.</param>
+        /// <param name="logger">Logger used to display warnings and errors.</param>
+        private void ReplayWarningsAndErrors(LockFile lockFile, ILogger logger)
+        {
+            var logMessages = lockFile?.LogMessages ?? Enumerable.Empty<IAssetsLogMessage>();
+
+            // Replay the log messages only if there were restore errors, because the assets file will not be committed.
+            // If there were only warnings then those are written to assets file and committed. The design time build will replay them.
+            if (logMessages.Any( m => m.Level == LogLevel.Error))
+            {
+                foreach (var logMessage in logMessages)
+                {
+                    var restoreLogMessage = new RestoreLogMessage(logMessage.Level, logMessage.Code, logMessage.Message)
```

Comment:
just curious, these look like the same data structures (in respect to properties) - what's the difference between them?